doc/nrf/security/tfm.rst

```diff
@@ -254,38 +266,447 @@ We will decrease the size of the (optional) ``mcuboot_pad`` partition and thus t
       address: 0x8000
       size: 0x4000
 
+Analyzing ``tfm_secure`` partition size
+=======================================
+
+The size of the ``tfm_secure`` partition can be analyzed from the build output.
+
+  .. code-block:: console
+
+     [71/75] Linking C executable bin/tfm_s.axf
+     Memory region   Used Size  Region Size  %age Used
+        FLASH:       31972 B       256 KB     12.20%
+        RAM:         4804 B        88 KB      5.33%
+
+The example above shows that the ``tfm_secure`` partition :kconfig:option:`CONFIG_PM_PARTITION_SIZE_TFM` is set to 256 kB and the TF-M binary uses 32 kB of the available space.
+Similarly the ``tfm_secure`` partition :kconfig:option:`CONFIG_PM_PARTITION_SIZE_TFM_SRAM` is set to 88 kB and the TF-M binary uses 5 kB of the available space.
+This information can be used to optimize the size of the TF-M, as long as it is within the alignment requirements explained in the previous section.
+
+Tools for analyzing the ``tfm_secure`` partition size
+-----------------------------------------------------
+
+TF-M build system provides ``tfm_ram_report`` and ``tfm_rom_report`` targets for analyzing the memory usage of the TF-M partitions inside ``tfm_secure`` partition.
+
+With |NCS|, the following commands can be used to generate the reports:
+
+  .. code-block:: console
+
+    west build -p -b nrf9151dk/nrf9151/ns samples/tfm/tfm_hello_world
+    cd build/tfm_hello_world
+    ninja tfm_ram_report
+    ninja tfm_rom_report
+
+Reports can be used to analyze the memory usage of the different TF-M partitions and how changing the Kconfig options affects the memory usage.
+For more information about the ``tfm_ram_report`` and ``tfm_rom_report`` targets, refer to the :ref:`tfm_build_system` documentation.
+
+TF-M Partitions
+===============
+
```

Comment:
Have some introductory text after a title. It is not a good practice to have subsequent titles without text in between.